### ISSUE TITLE
Convergence difficulties with pressure inlet BC in a 3D pipe

### ISSUE THREAD

**Issue by jacktat** [mikaem/Oasis#32]
Hello everyone,

I am experiencing difficulties reaching convergence with the Oasis solver with a problem I am trying to solve. I have a tube with a 6mm diameter with a fluid flowing through it which has a density of rho = 1000, and a dynamic viscosity of 5e-3 (blood). The boundary conditions which I have prescribed are the noslip condition on the walls, pressure is 0 at the outlet, and a constant pressure of 200 Pa at the inlet. I have tried ramping up to the desired pressure drop of 200 Pa, however this also doesn't converge. 
In the Oasis paper referring to equation (12) it is mentioned that “The boundary integral can be neglected for all parts of the domain where the velocity is prescribed.” I was wondering what happens to the boundary integral when pressure is prescribed and if this may be the caues of divergence, and if someone maybe has expierence in prescribing pressure using the Oasis solver. 
![oasis_boundary_integral](https://user-images.githubusercontent.com/60227728/98707745-ac64b800-2380-11eb-8415-7e38c28f3bd0.png)

Many thanks in advace.
Cheers,
Jack

**Comment by mikaem** [comment]
Is this a steady state problem? For steady state I strongly recommend using a coupled solver. If not, then note that equations are solved using Neumann on velocity if there is no Dirichlet boundary conditions set. That is, we do not include the ds-integral containing the velocity gradient on the boundaries, see, e.g., [here](https://github.com/mikaem/Oasis/blob/75e5a93d1401dd95086cd59f178f2f756677da05/oasis/solvers/NSfracStep/IPCS.py#L35).

**Comment by jacktat** [comment]
Thanks for your response. 
I'm sorry that I was not clear in my original post, I am solving a time dependent problem. 
I'm still having slight trouble understanding your reply. I thought phi in equation (12) was the pressure correction: "φ=pn−1/2−p∗ is a pressurecorrection and p∗is a tentative pressure". 
So I thought that boundary integral contained a pressure gradient, not a velocity gradient, is this correct? 

Cheers,
Jack

**Comment by mikaem** [comment]
Yes, the equation you show is for a pressure correction. I just ment to say that if you use a Dirichlet boundary condition on pressure, then you cannot at the same time use Dirichlet for velocity. So since we are not including the last term in Eq. (11), we are in fact using a Neumann boundary condition for velocity.

The boundary term you circle will be zero nevertheless, because you know the pressure (Dirichlet). There is no unknown on the boundary. Most textbooks would write that the testfunction is zero on a Dirichlet boundary.
